app/jobs/concerns/import_job.rb

```diff
@@ -53,8 +59,18 @@ def validate_headers(full_path)
 
   def import_records(filename)
     raise "No input file specified" unless filename
-    @stats = CsvImporter.import(filename, category_model.name.underscore.humanize.titleize, @processed_file.id)
-    log(stats, :info)
+    csv_importer = CsvImporter.new(filename, category_model.name.underscore.humanize.titleize, @processed_file.id)
+    csv_importer.call
+
+    if csv_importer.errored?
+      @error_details = csv_importer.error_details
+      log(error_details, :info)
+      return false
+    end
+
+    @job_stats = csv_importer.stats
+    log(job_stats, :info)
+    true
```

Comment:
Would this be better placed in the `else` block of the code above? I noticed that the code above short circuits and then this would never be reached. I think this is primarily a preference so I think it is fine either way.

Reply:
I like the way it's currently structured - I feel like putting everything else into an `else` after a guard `if` can make the code harder to read because you have to remember why the code is nested.